virttest/utils_net.py

```diff
@@ -3390,17 +3390,17 @@ def get_ip_address_by_interface(ifname, ip_ver="ipv4", linklocal=False):
         try:
             if linklocal:
                 return [a['addr'] for a in addr
-                        if a['addr'].lower().startswith(linklocal_prefix)][0]
+                        if a['addr'].lower().startswith(linklocal_prefix)][0].split("%")[0]
             else:
                 return [a['addr'] for a in addr
-                        if not a['addr'].lower().startswith(linklocal_prefix)][0]
+                        if not a['addr'].lower().startswith(linklocal_prefix)][0].split("%")[0]
         except IndexError:
             logging.warning("No IP address configured for "
-                            "the network interface %s !", ifname)
+                            "the network interface %s !" % ifname)
```

Comment:
It's not a typo.
Please refer to the documentation of the `logging` library.

Reply:
I see,it's not a syntax error, but the output should have the variable value. If not update it, the output contains \"%s\". It's not friendly, so need to update it. thanks.

e.g.
before updated:
No IP address configured for the network interface %s !  switch

after updated:
No IP address configured for the network interface  switch !

Reply:
```python
Python 3.7.1 (v3.7.1:260ec2c36a, Oct 20 2018, 14:57:15) [MSC v.1915 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import logging
>>> ifname = "switch"
>>> logging.error("Network interface: %s" % ifname)
ERROR:root:Network interface: switch
>>> logging.error("Network interface: %s", ifname)
ERROR:root:Network interface: switch
```

Reply:
OK, I will double check my debug logs. thanks.